analyze.go

```diff
@@ -2186,6 +2186,9 @@ var directives = map[string][]uint{
 		ngxHTTPMainConf | ngxHTTPSrvConf | ngxHTTPLocConf | ngxConfTake1,
 		ngxStreamMainConf | ngxStreamSrvConf | ngxConfTake1,
 	},
+	"js_periodic": {
+		ngxHTTPLocConf | ngxHTTPLifConf | ngxHTTPLmtConf | ngxConfTake1234,
```

Comment:
Can also be used in stream context. 

- [nginx/ngx_http_js_module.c#L336](https://github.com/nginx/njs/blob/5a384fc044e753eacc695b534a856e895b892780/nginx/ngx_http_js_module.c#L336)
- [nginx/ngx_stream_js_module.c#L192](https://github.com/nginx/njs/blob/5a384fc044e753eacc695b534a856e895b892780/nginx/ngx_stream_js_module.c#L192)

Reply:
More observations:

- From the source it doesn't look like it can be in an `if` or `limit_except` block
- `js_periodic` takes "zero or more arguments"
- In the stream context the directive goes into a `server` block.

Reply:
http://nginx.org/en/docs/http/ngx_http_js_module.html#js_periodic says:
Context: | location, if in location, limit_except

Reply:
docs also do not say first arg. (session object) are not optional, but the rest are?